src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/Helpers.cs

```diff
@@ -91,6 +91,21 @@ public static bool IsSliceLikeMethod(IMethodSymbol method)
                IsSliceFirstParameter(method.OriginalDefinition.Parameters[0]) &&
                IsSliceSecondParameter(method.OriginalDefinition.Parameters[1]);
 
+        /// <summary>
+        /// Look for methods like "SomeType MyType.Slice(int start)".  Note that the
+        /// name of the parameter is checked to ensure it is appropriate slice-like.
+        /// This name was picked by examining the patterns in the BCL for slicing members.
+        /// </summary>
+        public static bool IsSliceLikeMethodWithOneArgument(IMethodSymbol method)
+            => method != null &&
+               IsPublicInstance(method) &&
+               method.Parameters.Length == 1 &&
+               // From: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md#decisions-made-during-implementation
+               //
+               // When looking for the pattern members, we look for original definitions, not
+               // constructed members
```

Comment:
just make this a normal method.  breaking up an expression with complex comments like this just makes things too hard to read.

Reply:
Fixed.